front/app/containers/ProjectsShowPage/ProjectCTABar/index.tsx

```diff
@@ -57,20 +64,42 @@ const ProjectCTABar = ({ projectId }: ProjectCTABarProps) => {
     return () => window.removeEventListener('scroll', handleScroll);
   }, [isSmallerThanTablet, sticksToBottom]);
 
-  const participationMethod = project
-    ? getParticipationMethod(project.data, phases?.data)
-    : undefined;
-
-  if (!project || !participationMethod) {
+  if (!project || !phases) {
     return null;
   }
 
-  const BarContents = getMethodConfig(participationMethod).renderCTABar({
-    project: project.data,
-    phases: phases?.data,
-  });
-
-  if ((sticksToBottom || sticksToTop) && portalElement) {
+  const CTABars: { [method in ParticipationMethod]: JSX.Element | null } = {
+    ideation: <IdeationCTABar project={project.data} phases={phases.data} />,
+    proposals: <IdeationCTABar project={project.data} phases={phases.data} />,
+    native_survey: (
+      <NativeSurveyCTABar project={project.data} phases={phases.data} />
+    ),
+    information: <EventsCTABar project={project.data} phases={phases.data} />,
+    survey: (
+      <EmbeddedSurveyCTABar project={project.data} phases={phases.data} />
+    ),
+    voting: <VotingCTABar project={project.data} phases={phases.data} />,
+    poll: <PollCTABar project={project.data} phases={phases.data} />,
+    volunteering: (
+      <VolunteeringCTABar project={project.data} phases={phases.data} />
+    ),
+    document_annotation: (
+      <DocumentAnnotationCTABar project={project.data} phases={phases.data} />
+    ),
+  };
+
+  const participationMethod = getParticipationMethod(project.data, phases.data);
+  const CTABar = participationMethod ? CTABars[participationMethod] : null;
+
+  if (
+    (sticksToBottom || sticksToTop) &&
+    // We need to check that CTABar is defined because, at the time of writing,
+    // we rely on on the id of the portal below to determine how far to push down the filters.
+    // CTABar needs to be defined before it makes sense to render the portal (and push down the filters).
+    // Comment id: ccf2e3f. Search for this id to find related comments.
+    CTABar &&
```

Comment:
@amanda-anderson I overlooked something. `CTABar` is not resolved until it is rendered. It'll be the React element object here, even if it'll resolve to null when we render, so the CTABar check here will currently always pass. 🙃 When I've fixed it, we can have a quick call if you want to explain.

Reply:
Clear! No need for a call for me - makes sense :)